config/localTemplate.go

```diff
@@ -452,6 +452,10 @@ type Local struct {
 	// MaxAcctLookback sets the maximum lookback range for account states,
 	// i.e. the ledger can answer account states questions for the range Latest-MaxAcctLookback...Latest
 	MaxAcctLookback uint64 `version[23]:"4"`
+
+	// MaxAPIBoxPerApplication defines the maximum total number of boxes per application that will be returned
+	// in GetApplicationBoxes REST API responses.
+	MaxAPIBoxPerApplication uint64 `version[24]:"10000"`
```

Comment:
(looking at the arithmetic more closely, I've amended the previous 5000 recommendation, to the much more radically smaller 500...)

Reply:
@tzaffi Certainly there's some subjectivity in the configuration here, but I don't follow the recommendation.

* Box names are <= 64 bytes.  With the configured limit (10,000), the max response size is 625KB.
* 625KB does _not_ strike me as an _unreasonably_ large default.  If anything, I imagined advocating for a larger value.  In the absence of a strong rationale, I vote to keep as is.

Reply:
I don't have a strong opinion here. I will keep this discussion open for bit longer to see if anyone else chimes in.

Reply:
I'd be more inclined to size it with respect to the limit imposed for for getting resources. `MaxAPIResourcesPerAccount` is 100,000. (10x bigger).  I'm not sure exactly what gets returned (probably not program bytecode?), but certainly an app can have 8k in global state, so boxes are at most 4x bigger than that.  I think 10,000 is safe.

Reply:
Oh man.  This limit limits the number of box _names_ but the other limit limits the number of resources that get expanded under an account.  The sum of names is way _smaller_ than the size of these expanded resources.  I think I prefer to take out this new constant, and use the existing one. Or make them both 100,000.

Reply:
>  I think I prefer to take out this new constant, and use the existing one. Or make them both 100,000.

* Given these options, I vote to increase `MaxAPIBoxPerApplication` to 100,000.
* Rationale:  
  * Using a dedicated configuration parameter is less likely to result in confusion than relying on the existing constant.  
  * It seems reasonable to me that we'd like a server-side limiting function rather than removing the constant.

Hearing no objection by EOD, I'll make the change.

Reply:
I opened https://github.com/algorand/go-algorand/pull/4541 to address the feedback.  Let's resolve once the child PR is merged.